include/pops/power_law_kernel.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * PoPS model - power law dispersal kernel
+ *
+ * Copyright (C) 2015-2020 by the authors.
+ *
+ * Authors: Margaret Lawrimore malawrim ncsu edu
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#ifndef POPS_POWER_LAW_KERNEL_HPP
+#define POPS_POWER_LAW_KERNEL_HPP
+
+#include "kernel_types.hpp"
+#include "raster.hpp"
+
+#include <random>
+
+namespace pops {
+
+using std::pow;
+
+/*! Dispersal kernel for power law distribution
+ *  class utilized by RadialKernel and DeterministicKernel
+ */
+class PowerLawKernel
+{
+protected:
+    double xmin;
+    double alpha;
+
+public:
+    PowerLawKernel(double xm, double a) : xmin(xm), alpha(a) {}
+
+    /*!
+     *  Returns random value from power law distribution
+     *  Used by RadialKernel to determine location of spread
+     *  @param generator uniform random number generator
+     *  @return value from power law distribution
+     */
+    template<class Generator>
+    double random(Generator& generator)
+    {
+        std::uniform_real_distribution<double> distribution(0.0, 1.0);
+        double x = distribution(generator);
+        // return pow(x, (1.0 / (-alpha_ + 1.0))) * xmin_;
+        // since distribution is 0 to 1 xmin needs to be greater than 0
+        if (xmin <= 0) {
+            xmin = 0.01;
+        }
```

Comment:
This is something to resolve once in the constructor for all functions and function calls.

Is there any compelling reason to choose 0.01 and not 0.001 or anything else? You could also throw std::invalid_argument. We usually do nice user friendly checks in the interfaces (rpops, r.pops.spread) as we are closer to what user may want or expect and here we just go with exceptions to avoid surprising results.